R/get_datagrid.R

```diff
@@ -936,6 +924,31 @@ get_datagrid.datagrid <- get_datagrid.visualisation_matrix
     )
   }
 
+  # find variables that were coerced on-the-fly
+  terms <- find_terms(x, flatten = TRUE)
+  factors <- grepl("^(as\\.factor|as_factor|factor|as\\.ordered|ordered)\\((.*)\\)", terms)
+  if (any(factors)) {
+    factor_expressions <- lapply(terms[factors], .str2lang)
+    cleaned_terms <- vapply(factor_expressions, all.vars, character(1))
+    for (i in cleaned_terms) {
+      if (is.numeric(data[[i]])) {
+        attr(data[[i]], "factor") <- TRUE
+      }
+    }
+    attr(data, "factors") <- cleaned_terms
+  }
+  logicals <- grepl("^(as\\.logical|as_logical|logical)\\((.*)\\)", terms)
+  if (any(logicals)) {
+    logical_expressions <- lapply(terms[logicals], .str2lang)
+    cleaned_terms <- vapply(logical_expressions, all.vars, character(1))
+    for (i in cleaned_terms) {
+      if (is.numeric(data[[i]])) {
+        attr(data[[i]], "logical") <- TRUE
+      }
+    }
+    attr(data, "logicals") <- cleaned_terms
+  }
+
   data
```

Comment:
@vincentarelbundock here's the code, we could add it to the new `get_data()` function, like at the end of `.get_data_from_environment()`.

Reply:
Thanks for the pointer! Feels like you probably had the right intuition here. The vast majority of people who call `get_data()` won't need this, so it probably makes sense to pay the overhead only when it is useful: when building a grid.

Reply:
Just wanted to confirm that I'm fine without this. Thanks again!